contents/tutorial/coding-java-junit/track.yml

```
@@ -0,0 +1,22 @@
+type: coding
+main: src/main/java/App.java
+editable:
+  - src/main/java/App.java
+readonly:
+  - src/test/java/AppTest.java
+  - src/test/java/Util.java
+  - pom.xml
+test: >
+  mvn test;
+  metatest do echo then expand junit target/surefire-reports/TEST-AppTest.xml;
```

```suggestion
build: javac src/main/java/App.java && mvn test-compile -q
test: >
  mvn test -q;
  metatest do echo then expand junit target/surefire-reports/TEST-AppTest.xml;
```

こうしましょう。

- -q オプションにより不要な出力を減らします。
- build を指定することで、ビルド失敗時にテストが実行されないようにします。
    - javac src/main/java/App.java は App.java 内でのコンパイルエラーを即座にユーザーに伝えるためです(mavenの実行は遅い)。
    - mvn test-compile -q はテストファイルを含むコンパイルエラーを検出するためです。例えば App.java に class App { } のみが書かれているとき、 App.java のみのコンパイルは成功しますが、テストファイルを含めたコンパイルには失敗します。

---

metatestではなく、gradleを使用するのでresoled